### PR TITLE
Shim for functional form of mktemp

### PR DESCRIPTION
**Pull request by malmaud**

### CODE REVIEW

**Comment by simonster** [comment]
I think (hope) you mean a shim, not a shiv...

**Comment by malmaud** [comment]
i was playing a videogame at the same time i submitted this

File: src/Compat.jl [hunk 1]
```diff
@@ -511,6 +511,27 @@ else
...
+        (tmp_path, tmp_io) = mktemp()
+        try
+            fn(tmp_path, tmp_io)
+        finally
+            close(tmp_io)
+            rm(tmp_path)
+        end
+    end
+
+    function mktempdir(fn::Function)
```
Comment (tkelman) [inline src/Compat.jl]: needs to be exported for old versions

the recursive rm might not be supported on 0.3

**Comment by malmaud** [comment]
OK, this is green now. Base exports `mktemp` and `mktempdir` as early as .3, so I think this is correct now on the export front.

**Comment by malmaud** [comment]
Good to merge?

**Comment by tkelman** [comment]
Oh, needs a readme update.

**Pull request merged**